### PR TITLE
Feat/add note

### PR DESCRIPTION
**Pull request by bhavyaj12**
### Added context, reducer for note state, now logged in user can add notes

- Create Note form added
- Display user notes 
- To check pls login using guest credentials

### What not to check

- Change color
- Edit Note
- Date added
- Other features (trash, archive etc)

### CODE REVIEW

File: src/components/NoteItems/NoteCard.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  EditRoundedIcon
+} from "../../components";
+import "./note-card.css";
+
+const NoteCard = ({ note }) => {
+  const { title, content } = note;
+  return (
+    <div className="note-container flex-col p-4 m-4">
+      <PushPinRoundedIcon className="pin-icon" />
+      <p className="txt-small note-label my-2">Label</p>
```
Comment (sashaboi) [inline src/components/NoteItems/NoteCard.jsx:15]: Here in the styling, if the title section is pulled up into a 3 item grid along with the label and pin , might help use space better? 
do try in later versions if UI guidelines permit.

File: src/components/NoteItems/NoteCard.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  PushPinRoundedIcon,
+  PaletteRoundedIcon,
+  EditRoundedIcon
+} from "../../components";
+import "./note-card.css";
+
+const NoteCard = ({ note }) => {
+  const { title, content } = note;
+  return (
+    <div className="note-container flex-col p-4 m-4">
```
Comment (sashaboi) [inline src/components/NoteItems/NoteCard.jsx:13]: image
Here, the red space seems to be creeping in , and could be used up ? suggestion for future.

File: src/components/NoteItems/AddNote.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        onChange={(e) => setNote({ ...note, title: e.target.value })}
+      />
+      <textarea
+        className="add-note-content p-3"
+        placeholder="Enter note..."
+        value={note.content}
+        onChange={(e) => setNote({ ...note, content: e.target.value })}
+      ></textarea>
+      <div className="add-note-footer flex-row my-3">
+        <button type="submit" className="button btn-solid button-primary">Save Note</button>
```
Comment (sashaboi) [inline src/components/NoteItems/AddNote.jsx:47]: white to pink contrast ratio do check , seems to be below three, which is usual guideline.

File: src/components/NavTop/NavTop.jsx [hunk 4]
```diff
@@ -33,20 +33,22 @@ const NavTop = () => {
...
+                  <Link to="/addnote" className="btn">
+                    Add Note
+                  </Link>
+                </li>
+                <li className="notes-nav-item">
+                  <button
+                    className="btn btn-logout"
+                    onClick={() => signOutFunc(setAuth)}
+                  >
+                    Logout
```
Comment (sashaboi) [inline src/components/NavTop/NavTop.jsx:48]: the mouseover half underline feature looks so cool ! 
does the logout button need that too though? hope its intentional.

**Review by sashaboi — APPROVED** [review]
LOVE the UI, the very thin vertical line on the sidebar is muah chefs kiss. 
Good to merge!
I have made a few suggestions, do go through when time permits.

forgot to add one more thing, the home page cta button should have a link to take you to take notes, could make the job of reviewer much easier :)

**Review by Balakumar3018 — APPROVED** [review]
Functionality is working as expected. Good to merge

**Pull request merged**